[MRG] update JOSS paper for v4

ctb opened the pull request:
Per https://github.com/openjournals/joss/issues/423, we are planning to make a new submission to JOSS for sourmash v4.x. The first paper from 2016 is [here](https://joss.theoj.org/papers/10.21105/joss.00027).

Fixes #1321.

Ref https://github.com/dib-lab/sourmash/issues/622, https://github.com/dib-lab/sourmash/issues/444.

- [x] update authors per https://github.com/dib-lab/sourmash/issues/1367
- [x] fix `@@@` in paper.md YAML header
- [x] review [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html) and think about what to add, if anything.
- [x] add funding acks - Moore esp
- [ ] let's fix citation issue finally https://github.com/dib-lab/sourmash/issues/511

ctb committed:
update JOSS paper for v4

ctb committed:
more author update

ctb committed:
update text

ctb committed:
Merge branch 'latest' into update/joss

ctb committed:
Merge branch 'latest' of github.com:dib-lab/sourmash into update/joss

ctb committed:
minor text update

ctb committed:
update citations

ctb commented:
Does anyone have any comments on the text? @luizirber @bluegenes in particular.

Also, I can't figure out how to get the references to work properly using https://whedon.theoj.org/, but 🤷

bluegenes commented on paper.md:
```diff
@@ -1,32 +1,109 @@
 ---
-title: 'sourmash: a library for MinHash sketching of DNA'
+title: 'sourmash: a tool to quickly search, compare, and analyze genomic and metagenomic data sets'
```
"genomic and metegenomic" ignores transcriptome/metatranscriptome data, etc. But since we have mostly tested/developed workflows for genomic/metagenomic data, this is probably fine?

ctb replied:
yeah, that was basically my thinking here :)

bluegenes commented on paper.md:
```diff
@@ -1,32 +1,109 @@
-build phylogenetic trees [@ondov2015fast].
+using a Jaccard index [@Ondov:2015].  MinHash sketches can be used
+to identify samples, find similar samples, identify data sets with
+shared sequences, and build phylogenetic trees
```
```suggestion
shared sequences, and build phylogenetic trees.
```

bluegenes commented on paper.md:
```diff
@@ -1,32 +1,109 @@
+
+Since sourmash v1 was released in 2016 [@Brown:2016], sourmash has expanded
+to support new database types and many more command line functions.
+In particular, sourmash now has robust support for both Jaccard similarity
+and containment calculations, which supports robust analysis and comparison
+of large metagenomic samples.
```
```suggestion
In particular, sourmash now has robust support for both Jaccard similarity
and containment calculations, which enables analysis and comparison of data sets
of different sizes, including large metagenomic samples.
```

one option for avoiding `robust` and `support` twice in same sentence :)

bluegenes commented on paper.md:
```diff
@@ -1,32 +1,109 @@
+
+# Statement of Need
+
+Large collections of genomes, transcriptomes, and raw sequencing data
+sets are readily available in biology, and the field needs lightweight
+computational methods for searching and summarizing the content of
+both public and private collections. sourmash provides a flexible set
+of programmatic functionality for this purpose, together with a robust
+and well-tested command-line interface. It has been used in over 70
+publications (based on citations of @Brown:2016) and it continues
+to expand in functionality.
```
beautiful.

bluegenes reviewed (approved):
Made some suggestions. Otherwise looks good to me!

ctb committed:
Update paper.md

Co-authored-by: Tessa Pierce Ward <bluegenes@users.noreply.github.com>

ctb committed:
Update paper.md

Co-authored-by: Tessa Pierce Ward <bluegenes@users.noreply.github.com>

ctb committed:
Merge branch 'latest' of github.com:dib-lab/sourmash into update/joss

ctb committed:
Merge branch 'update/joss' of github.com:dib-lab/sourmash into update/joss

ctb commented:
@luizirber if there aren't any concerns on your part, I'll update the authors one more time and submit to JOSS in the next week or two.

bluegenes commented:
@ctb @luizirber I'd like to update this further for ANI (adding David and Mahmudur + some ANI text), and then submit to JOSS (after 4.4 is out).

I assume there are some other authors we may also need to add or author ordering to change, given the time since this was updated - @ctb is this something you could help with or guide me through?

ctb committed:
Merge branch 'latest' of https://github.com/sourmash-bio/sourmash into update/joss

ctb committed:
add joss github action

ctb commented:
Main thing remaining to be done: fix all the missing information in the authors (search for `@@@` in paper.md)

ctb commented:
@bluegenes I think we should pull in `CITATION.cff` and maybe `codemeta.json` from https://github.com/sourmash-bio/sourmash/pull/837/ and update them on this PR; seem good? I can do some light editing if you can proof :)

luizirber commented:
> Main thing remaining to be done: fix all the missing information in the authors (search for `@@@` in paper.md)
>
> @bluegenes I think we should pull in `CITATION.cff` and maybe `codemeta.json` from #837 and update them on this PR; seem good? I can do some light editing if you can proof :)

I think the author list in pyproject.toml is the most updated (with all the ORCID): https://github.com/sourmash-bio/sourmash/blob/059d98853b0b390c9d69832f132e0812be70448c/pyproject.toml#L14L49

bluegenes committed:
[MRG] update JOSS for sourmash 4.4 (#2006)

draft for sourmash 4.x.

Co-authored-by: C. Titus Brown <titus@idyll.org>

bluegenes commented:
@ctb, should I reorder paper author list to be alphabetical except for first/last authors, as in `pyproject.toml`? Or keep custom ordering and insert authors based on # contributions (per gh contribution page)?

bluegenes committed:
Merge branch 'latest' into update/joss

bluegenes committed:
authorship order

bluegenes committed:
add auths; reorder to alphabetical except first/last

bluegenes commented:
checking authorship as of 4.6.1:

`git shortlog --summary --numbered`

```
  1047  C. Titus Brown
   224  Luiz Irber
    77  dependabot[bot]
    38  Tessa Pierce Ward
    34  Laurent Gautier
    22  Keya Barve
    17  Tim Head
    10  Mohamed Abuelanin
     8  Olga Botvinnik
     8  Taylor Reiter
     7  Erik Young
     7  Connor Tiffany
     5  Pranathi Vemuri
     4  Tessa Pierce
     4  dependabot-preview[bot]
     3  Hannah Eve Houts
     3  Daniel Standage
     3  S. Joshua Swamidass
     3  N. Tessa Pierce-Ward
     3  brooksph
     2  David Koslicki
     2  Titus Brown
     2  Camille Scott
     2  ccbaumler
     2  Harriet Alexander
     1  Phillip Brooks
     1  Andreas Sjödin
     1  Arfon Smith
     1  Connor T. Skennerton
     1  Daniel Dsouza
     1  Fabian Klötzl
     1  Ivan Ogasawara
     1  Jason Stajich
     1  Katrin Leinweber
     1  Marisa Lim
     1  NapsterInBlue
     1  Peter Cock
     1  Abhishek Anant
     1  Ricky Lim
     1  ljcohen
     1  pyup.io bot
     1  swamidass
```



ref #622

bluegenes committed:
spacing

bluegenes committed:
alpha ordering

bluegenes committed:
comment @ char

bluegenes committed:
ask instead

bluegenes commented on paper.md:
```diff
@@ -1,32 +1,150 @@
+ - name: David Koslicki
+   orcid: 0000-0002-0640-954X
+   affiliation: Pennsylvania State University
+ - name: Katrin Leinweber
+   orcid: 0000-0001-5135-5758
+   #affiliation: @@@
```
hi @katrinleinweber, I am updating the list of sourmash authors to release a JOSS paper for version 4.x. Can you please reply with your desired affiliation? Thanks!

(Of course, also please let me know if you'd prefer not to be included as an author.)

katrinleinweber replied:
Thanks for the ping, @bluegenes! Not including me seems better: https://github.com/sourmash-bio/sourmash/pull/2473

I'm glad that I could help with a small thing and wish you all the best 👋

bluegenes commented on paper.md:
```diff
@@ -1,32 +1,150 @@
+ - name: Lisa K. Johnson
+   orcid: 0000-0002-3600-7218
+   affiliation: 10x Genomics
+ - name: Fabian Klötzl
+   orcid: 0000-0002-6930-0592
+   #affiliation: @@@
```
Hi @kloetzl, I am updating the list of sourmash authors to release a JOSS paper for version 4.x. Can you please reply with your desired affiliation? Thanks!

(Of course, also please let me know if you'd prefer not to be included as an author.)

kloetzl replied:
Hi, my affiliation would be `MPI for Evolutionary Biology`. Thanks for including me.

bluegenes commented on paper.md:
```diff
@@ -1,32 +1,150 @@
+ - name: Peter Cock
+   orcid: 0000-0001-9513-9993
+   #affiliation: @@@
+ - name: Daniel Dsouza
+   orcid: 0000-0001-7843-8596
+   #affiliation: @@@
```
hi @dsouzadaniel, I am updating the list of sourmash authors to release a JOSS paper for version 4.x. Can you please reply with your desired affiliation? Thanks!

(Of course, also please let me know if you'd prefer not to be included as an author.)

bluegenes commented on paper.md:
```diff
@@ -1,32 +1,150 @@
+ - name: Daniel Dsouza
+   orcid: 0000-0001-7843-8596
+   #affiliation: @@@
+ - name: Laurent Gautier
+   orcid: 0000-0003-0638-3391
+   #affiliation: @@@
```
hi @lgautier, I am updating the list of sourmash authors to release a JOSS paper for version 4.x. Can you please reply with your desired affiliation? Thanks!

(Of course, also please let me know if you'd prefer not to be included as an author.)

lgautier replied:
I believe that this is the orcid you are looking for: 0000-0002-5670-4512.

bluegenes replied:
got it, thanks!

bluegenes commented on paper.md:
```diff
@@ -1,32 +1,150 @@
+ - name: Phillip Brooks
+   orcid: 0000-0003-3987-244X
+   affiliation: Ancestry
+ - name: Peter Cock
+   orcid: 0000-0001-9513-9993
+   #affiliation: @@@
```
Hi @peterjc, I am updating the list of sourmash authors to release a JOSS paper for version 4.x. Can you please reply with your desired affiliation? Thanks!

(Of course, also please let me know if you'd prefer not to be included as an author.)

peterjc replied:
Looking back my only contribution to date was a tiny cosmetic documentation fix, so I'll decline - but thank you for asking.

bluegenes replied:
got it. Thank you for the response and the contribution!

bluegenes commented on paper.md:
```diff
@@ -1,32 +1,150 @@
+ - name: Harriet Alexander
+   orcid: 0000-0003-1308-8008
+   affiliation: Woods Hole Oceanographic Institute
+ - name: Abhishek Anant
+   orcid:  0000-0002-5751-2010
+   #affiliation: @@@
```
hi @itsabhianant, I am updating the list of sourmash authors to release a JOSS paper for version 4.x. Can you please reply with your desired affiliation? Thanks!

(Of course, also please let me know if you'd prefer not to be included as an author.)

bluegenes commented on paper.md:
```diff
@@ -1,32 +1,150 @@
+ - name: Ricky Lim
+   orcid: 0000-0003-1313-7076
+   #affiliation: @@@
+ - name: Ivan Ogasawara
+   orcid: 0000-0001-5049-4289
+  #  affiliation: @@@
```
hi @xmnlab, I am updating the list of sourmash authors to release a JOSS paper for version 4.x. Can you please reply with your desired affiliation? Thanks!

(Of course, also please let me know if you'd prefer not to be included as an author.)

xmnlab replied:
It would be an honor to be listed there :)
I am very bad with this thing of affiliation, never know what to use :S if it could be something like community, maybe Open Science Labs and ArxLang, otherwise if it could be a scientific project, maybe EpiGraphHub and Mosqlimate. Let me know if you have any thoughts about that or any recommendation

bluegenes replied:
I realized we don't need to list an affiliation for JOSS, so perhaps I'll leave it blank and just link your ORCID? Let me know if you have a preference, thanks!

bluegenes commented on paper.md:
```diff
@@ -1,32 +1,150 @@
+ - name: Marisa Lim
+   orcid: 0000-0003-2097-8818
+   affiliation: 10x Genomics
+ - name: Ricky Lim
+   orcid: 0000-0003-1313-7076
+   #affiliation: @@@
```
hi @ricky-lim, I am updating the list of sourmash authors to release a JOSS paper for version 4.x. Can you please reply with your desired affiliation? Thanks!

(Of course, also please let me know if you'd prefer not to be included as an author.)

bluegenes commented on paper.md:
```diff
@@ -1,32 +1,150 @@
+ - name: Camille Scott
+   orcid: 0000-0001-8822-8779
+   affiliation: University of California, Davis
+ - name: Andreas Sjödin
+   orcid: 0000-0001-5350-4219
+   #affiliation: @@@
```
hi @druvus, I am updating the list of sourmash authors to release a JOSS paper for version 4.x. Can you please reply with your desired affiliation? Thanks!

(Of course, also please let me know if you'd prefer not to be included as an author.)

druvus replied:
My affiliation is Swedish Defence Research Agency (FOI), Umeå, Sweden

bluegenes replied:
got it, thanks!

bluegenes commented on paper.md:
```diff
@@ -1,32 +1,150 @@
+ - name: Andreas Sjödin
+   orcid: 0000-0001-5350-4219
+   #affiliation: @@@
+ - name: Connor Skennerton
+   orcid: 0000-0003-1320-4873
+   #affiliation: @@@
```
hi @ctSkennerton, I am updating the list of sourmash authors to release a JOSS paper for version 4.x. Can you please reply with your desired affiliation? Thanks!

(Of course, also please let me know if you'd prefer not to be included as an author.)

bluegenes commented on paper.md:
```diff
@@ -1,32 +1,150 @@
+ - name: Connor Skennerton
+   orcid: 0000-0003-1320-4873
+   #affiliation: @@@
+ - name: Jason Stajich
+   orcid: 0000-0002-7591-0020
+   #affiliation: @@@
```
hi @hyphaltip, I am updating the list of sourmash authors to release a JOSS paper for version 4.x. Can you please reply with your desired affiliation? Thanks!

(Of course, also please let me know if you'd prefer not to be included as an author.)

hyphaltip replied:
I'd have loved to contribute but my only code updates are around a PMID fix... so I don't think that rises to authorship here. thanks for being inclusive.

hyphaltip replied:
I'd have loved to contribute but my only code updates are around a PMID fix... so I don't think that rises to authorship here. thanks for being inclusive but I don't need to be listed as an author.

bluegenes replied:
got it, thanks for your contribution!

bluegenes commented on paper.md:
```diff
@@ -1,32 +1,150 @@
+ - name: Jason Stajich
+   orcid: 0000-0002-7591-0020
+   #affiliation: @@@
+ - name: Daniel Standage
+   orcid: 0000-0003-0342-8531.
+   #affiliation: @@@
```
hi @standage, I am updating the list of sourmash authors to release a JOSS paper for version 4.x. Can you please reply with your desired affiliation? Thanks!

standage replied:
My current affiliation is the National Bioforensic Analysis Center, Frederick MD. Thanks.

bluegenes replied:
added, thanks!

bluegenes commented on paper.md:
```diff
@@ -1,32 +1,150 @@
+ - name: Nick H
+   orcid: 0000-0002-1685-302X
+   #affiliation: @@@
+ - name: Tim Head
+   orcid: 0000-0003-0931-3698
+   #affiliation: @@@
```
hi @betatim, I am updating the list of sourmash authors to release a JOSS paper for version 4.x. Can you please reply with your desired affiliation? Thanks!

(Of course, also please let me know if you'd prefer not to be included as an author.)

betatim replied:
I now work for Nvidia.

However, I've not contributed to sourmash in a long time. I think this means my "retirement benefits" have been used up and I'm not an author anymore :)

bluegenes commented on paper.md:
```diff
@@ -1,32 +1,150 @@
+ - name: Laurent Gautier
+   orcid: 0000-0003-0638-3391
+   #affiliation: @@@
+ - name: Nick H
+   orcid: 0000-0002-1685-302X
+   #affiliation: @@@
```
hi @NapsterInBlue, I am updating the list of sourmash authors to release a JOSS paper for version 4.x. Can you please reply with your desired affiliation? Thanks!

(Of course, also please let me know if you'd prefer not to be included as an author.)

NapsterInBlue replied:
Echoing what others have wrote-- sincerely appreciate the offer, but I don't feel like I've contributed remotely enough for attribution.

In any case, I hope the new year's treating everyone well!

bluegenes replied:
got it - thanks for your contribution!

bluegenes committed:
rm unnecessary punctuation

ctSkennerton commented:
I think it’s safe to remove me from the author list. I haven’t contributed in a long time now.On Jan 23, 2023, at 13:57, Tessa Pierce Ward ***@***.***> wrote:﻿
@bluegenes commented on this pull request.



In paper.md:
> + - name: Connor Skennerton
+   orcid: 0000-0003-1320-4873
+   #affiliation: @@@

hi @ctSkennerton, I am updating the list of sourmash authors to release a JOSS paper for version 4.x. Can you please reply with your desired affiliation? Thanks!
(Of course, also please let me know if you'd prefer not to be included as an author.)

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

katrinleinweber committed:
Remove Katrin Leinweber due to insignificant contribution (#2473)

I only ran a script to generate:
https://github.com/sourmash-bio/sourmash/pull/562 😊

bluegenes committed:
Merge branch 'update/joss' of github.com:sourmash-bio/sourmash into update/joss

bluegenes committed:
upd orcid

bluegenes committed:
upd authorship

bluegenes committed:
rm empty affil

bluegenes committed:
add corr, eq info

bluegenes committed:
index affiliations

bluegenes commented:
Hi coauthors,

We're getting ready to submit this to JOSS! Can you please review your information and the manuscript text and provide any comments by April 21st? 

shortcut to files changed: https://github.com/sourmash-bio/sourmash/pull/1361/files
rendered pdf attached here.
[paper.pdf](https://github.com/sourmash-bio/sourmash/files/11115218/paper.pdf)

@luizirber @mr-eyes @halexand @itsabhianant @keyabarve @ccbaumler @olgabot @brooksph @dsouzadaniel @lgautier @mahmudhera @hehouts @johnsolk @kloetzl @dkoslicki @marisalim @ricky-lim @xmnlab @taylorreiter @camillescott @druvus @standage @swamidass @connor-reid-tiffany @pranathivemuri @erikyoung85 @ctb

mr-eyes commented:
Everything sounds great to me.

xmnlab commented:
just a question, I was listed without a affiliation: https://github.com/sourmash-bio/sourmash/pull/1361/files#diff-0403ef06adf405f7b310b4518bd6a3559854f54c61676f676ce9cbfee7172ab6R64
but in the pdf rendered shows affiliation 1 ... is that expected?

bluegenes committed:
fix no affil auths

bluegenes committed:
fix lg orcid

bluegenes committed:
fix idnt

bluegenes commented:
> just a question, I was listed without a affiliation: https://github.com/sourmash-bio/sourmash/pull/1361/files#diff-0403ef06adf405f7b310b4518bd6a3559854f54c61676f676ce9cbfee7172ab6R64 but in the pdf rendered shows affiliation 1 ... is that expected?

Thanks for catching that -- unexpected. Here's the updated version

[paper.pdf](https://github.com/sourmash-bio/sourmash/files/11124688/paper.pdf)

xmnlab commented:
> Thanks for catching that -- unexpected. Here's the updated version
> 
> [paper.pdf](https://github.com/sourmash-bio/sourmash/files/11124688/paper.pdf)

thanks @bluegenes ! thanks look good to me :)

bluegenes committed:
upd affil

bluegenes commented:
updated @marisalim @johnsolk to UC Davis affil to reflect time of work

[paper.pdf](https://github.com/sourmash-bio/sourmash/files/11130538/paper.pdf)

swamidass commented:
Thanks for including me. This is seems more like an abstract than a paper. But that isn't necessarily a problem for JOSS. 

My affiliation is: Associate Professor, Laboratory and Genomic Medicine, Washington University in St Louis

If there is a full scientific paper written at some point, there are some additional algorithms to consider benchmarking. Please reach out to me early enough on so that I can help with the design.

bluegenes commented:
> Thanks for including me. This is seems more like an abstract than a paper. But that isn't necessarily a problem for JOSS.
> 
> My affiliation is: Associate Professor, Laboratory and Genomic Medicine, Washington University in St Louis
> 

> If there is a full scientific paper written at some point, there are some additional algorithms to consider benchmarking. Please reach out to me early enough on so that I can help with the design.

Hi @swamidass,

Yes - the goal here is an updated general citation for the software. If you're interested in our larger paper efforts, please see our 2022 preprint, [Lightweight compositional analysis of metagenomes with FracMinHash and minimum metagenome covers](https://www.biorxiv.org/content/10.1101/2022.01.11.475838v2) and our 2019 use cases paper, [Large-scale sequence comparisons with sourmash](https://f1000research.com/articles/8-1006).

We would welcome any additional contributions or discussion on the [issue tracker](https://github.com/sourmash-bio/sourmash/issues)!

I will add your affiliation here.

thanks!

updated pdf: [paper.pdf](https://github.com/sourmash-bio/sourmash/files/11143552/paper.pdf)

bluegenes committed:
add swamidass afill

taylorreiter commented:
looks good to me!

mahmudhera commented:
Looks good. Thanks!

connor-reid-tiffany commented:
Looks good on my end.

On Thu, Apr 6, 2023, 10:43 AM Mahmudur Rahman Hera ***@***.***>
wrote:

> Looks good. Thanks!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/sourmash-bio/sourmash/pull/1361#issuecomment-1499411719>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHIC6DWZQEDDEHAAQSMXQDLW736ERANCNFSM4YSI5ANQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

dkoslicki commented:
Looks good to me as well!

halexand commented on paper.md:
```diff
@@ -1,32 +1,155 @@
+affiliations:
+ - name: University of California, Davis
+   index: 1
+ - name: Woods Hole Oceanographic Institute
```
Lokos good-- but should be Institution

bluegenes replied:
😂 I should have caught this - happens with Scripps all the time. Will fix!

bluegenes committed:
institution!

standage commented:
Hi @bluegenes, could we add the following (not super relevant but unfortunately still required) statement to the acknowledgements?

> Notice: This manuscript has been authored by BNBI under Contract
No. HSHQDC-15-C-00064 with the DHS. The US Government retains
and the publisher, by accepting the article for publication, acknowledges
that the USG retains a non-exclusive, paid-up, irrevocable, world-wide
license to publish or reproduce the published form of this manuscript,
or allow others to do so, for USG purposes. Views and conclusions
contained herein are those of the authors and should not be interpreted
to represent policies, expressed or implied, of the DHS.

bluegenes committed:
add dhs statement from @standage

bluegenes commented:
@ctb ready for review and merge

Let's revisit making citation guidelines clearer after pyopensci review?

ctb committed:
Merge branch 'latest' into update/joss
